src/pages/login/login.ts

```diff
@@ -14,11 +21,36 @@ import { NavController, NavParams } from 'ionic-angular';
 })
 export class LoginPage {
 
-  constructor(public navCtrl: NavController, public navParams: NavParams) {
+  private loginCreds : FormGroup;
+
+  constructor(public navCtrl: NavController, public navParams: NavParams, public _userService: UserProvider, private formBuilder: FormBuilder) {
+    this.loginCreds = this.formBuilder.group({
+      email: ['', Validators.compose([Validators.required,
+        Validators.pattern('^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+$')])],
+      password: ['', Validators.required],
+    });
   }
 
   ionViewDidLoad() {
     console.log('ionViewDidLoad LoginPage');
   }
 
+  login() {
+    this._userService.login(this.loginCreds.value)
+      .subscribe(
+        (res) => {
+          alert("you're logged in!")
+         this.toDashboard();
+        },
+        (err) => alert("Invalid credentials")
+      )
+  }
+
+  toRegisterPage() {
+    this.navCtrl.push(RegisterPage)
+  }
+
+  toDashboard() {
+    this.navCtrl.setRoot(DashboardPage);
```

Comment:
Nice, you used this correctly. During code review, lets remind people diff b/w `setRoot` and `push`